### PR TITLE
Fix setting PATH during linkage on windows-gnu

### PR DESCRIPTION
**Pull request by brson**
This makes the behavior almost exactly the same as before the VS2017 patch, except that on MSVC builds the host bin path is no longer added to PATH. I am not sure that's actually necessary on any platform.

r? @alexcrichton 

Fixes https://github.com/rust-lang/rust/issues/42422

### CODE REVIEW

**Comment by brson** [comment]
Oh actually I miststated - this is not like previously, now this is only frobbing PATH at all for linker invocation on windows, whereas before it was doing it on all platforms. Again we think it only matters on windows.

File: src/librustc_trans/back/link.rs [hunk 1]
```diff
@@ -141,17 +141,31 @@ pub fn build_link_meta(incremental_hashes_map: &IncrementalHashesMap) -> LinkMet
...
+// The third parameter is for env vars, used on windows to set up the
+// path for MSVC to find its DLLs, and gcc to find its bundled
+// toolchain
 pub fn get_linker(sess: &Session) -> (String, Command, Vec<(OsString, OsString)>) {
+    let envs = |linker: Option<&str>| -> Vec<(OsString, OsString)> {
+        if sess.target.target.options.is_like_windows {
+            // On windows-gnu (msvc is handled separately), we bundle
+            // gcc and other tools with the official releases. This is
+            // how the other tools that gcc itself needs are located.
+            vec![("PATH".into(), command_path(sess, linker.map(PathBuf::from)))]
```
Comment (ollie27) [inline src/librustc_trans/back/link.rs]: This doesn't seem right. The second parameter to `command_path` is added to the end of `%Path%` and before #42225 it was only used by MSVC.

Comment (brson) [inline src/librustc_trans/back/link.rs]: Ah yes. I suspected that path had no effect. I'll update.

**Comment by brson** [comment]
Ok, now the patch is written so that all non-msvc platforms set up the PATH during linking as previously. The `command_path` function only takes one argument now.

**Pull request merged**